src/panels/lovelace/components/hui-image.js

```diff
@@ -108,18 +110,35 @@ class HuiImage extends LocalizeMixin(PolymerElement) {
 
     const state = hass.states[this.entity];
     const unavailable = !isValidObject(state, ['state']);
+    const newState = unavailable ? 'unavailable' : state.state;
+
+    if (newState === this._currentState) return;
+    this._currentState = newState;
 
+    this._updateStateImage();
+    this._updateStateFilter(state, unavailable);
+  }
+
+  _updateStateImage() {
     if (!this.stateImage) {
-      this._imageClass = unavailable || STATES_OFF.includes(state.state) ? 'state-off' : '';
+      this._imageFallback = true;
       return;
     }
+    const stateImg = this.stateImage[this._currentState];
+    this._imageSrc = stateImg || this.image;
+    this._imageFallback = !stateImg;
+  }
 
-    const stateImg = !unavailable ? this.stateImage[state.state] : this.stateImage.unavailable;
+  _updateStateFilter(state, unavailable) {
```

Comment:
we should call it `stateObj`. Isn't it redundant to have both, stateObj and unavailable? If it's unav. stateObj should be undefined?

Reply:
Yeah it's a little redundant, but I already checked that it was an object in `_hassChanged()` so I figured I would just pass that to the method as well to avoid checking it again before trying to access `.state`. I can just check it in `_updateStateFilter()` too if you prefer it that way

Reply:
I suggest we get rid of the `unavailable` variable :+1:

Reply:
👍 done